### PR TITLE
InputStreamFromROOTFile changes for virtual crystals

### PR DESCRIPTION
**Pull request by KrisThielemans**
Closing #357

### CODE REVIEW

**Comment by KrisThielemans** [comment]
work in progress

**Comment by KrisThielemans** [comment]
This might be working now for the mCT with some limitations/caveats:
- although you can set the number of virtual crystals via the hroot file, you cannot set it to anything else than what `stir::Scanner` expects (due to a limitation in `Scanner`). Therefore, you have to make sure that the hroot file specifies the scanner name correctly (i.e. either "Siemens mCT" or "mCT").
- although the current [run_root_GATE.sh](https://github.com/UCL/STIR/blob/master/recon_test_pack/run_root_GATE.sh) passes, it does only checks on total counts, so I don't know if I broke anything!

Things to do therefore:
- [ ] modify `stir:Scanner` to allow changing the number of virtual crystals
- [ ] modify `run_root_GATE.sh` to do some tests on actual coincidence decoding
- [ ] test with mCT GATE output

File: src/IO/InputStreamFromROOTFile.cxx [hunk 1]
```diff
@@ -56,6 +66,11 @@ InputStreamFromROOTFile::set_defaults()
     low_energy_window = 0.f;
     up_energy_window = 1000.f;
     read_optional_root_fields=false;
+    crystal_repeater_x = -1;
+    crystal_repeater_y = -1;
+    crystal_repeater_z = -1;
```
Comment (robbietuk) [inline src/IO/InputStreamFromROOTFile.cxx:71]: Does this really need to be -1? STIR can only handle cases right now where z= 1.

**Comment by KrisThielemans** [comment]
could test ROOT interpretation in more detail by using for instance
```
INPUT_ROOT_FILE=test_PET_GATE.root EXCLUDE_SCATTERED=1 EXCLUDE_RANDOM=0 list_lm_events --coincidence 1 --time 0 --num-events-to-list 100 root_header.hroot 
```
A simple thing to do would be to pipe the output of this to a file with a version that is known to work, add the file to the `recon_test_pack` and in the test, compare the output. We'd have to do the same for the timing events.

File: src/IO/InputStreamFromROOTFile.cxx (hunk 1 above)
Comment (KrisThielemans) [reply src/IO/InputStreamFromROOTFile.cxx:71]: I've only moved this code from the original location in the derived classes, so no change there. I didn't check if there's actually any error checking or handling of defaults on those values. I agree it seems more logical to let them default to 1 but it doesn't seem high priority

**Comment by robbietuk** [comment]
I unlisted the same root file from a D690 scanner with `lm_to_projdata` from using master and this PR branch. I get the same number of events in both sinograms and subtracting one data file from the other gives a file of 0's. It appears that this PR does not affect the current unlisting code regarding unlisting root files in number of events or shape.

**Comment by KrisThielemans** [comment]
I've updated the doc and rebased on `master` as the history got a bit convoluted (and there was a conflict). should be good to go.

**Comment by KrisThielemans** [comment]
2 Travis jobs failed, nothing to do with us (networking error and unknown error). Merging.

**Pull request merged**